scanner/src/main/kotlin/experimental/ProvenanceTree.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.experimental
+
+import org.ossreviewtoolkit.model.KnownProvenance
+import org.ossreviewtoolkit.model.RepositoryProvenance
+
+/**
+ * This class contains information about a [root] provenance and all nested [subRepositories].
+ */
+data class ProvenanceTree(
```

Comment:
Sorry to go into a naming discussion again, but as this only applies to `RepositoryProvenance`, not to `ArtifactProvenance`, and as in the context of a repository the term "tree" is commonly used for the file tree, I'd propose a name like `NestedRepositoryProvenance`. That would mach the `nestedRepositories` from `Repository`.

Reply:
I went with `NestedProvenace` (see below).